api/src/main/java/org/openmrs/util/TimeZoneUtil.java

```diff
@@ -0,0 +1,99 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import org.joda.time.DateTime;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+import org.openmrs.User;
+import org.openmrs.api.context.Context;
+import org.openmrs.ui.framework.UiFrameworkConstants;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+import static org.joda.time.DateTimeZone.UTC;
+
+/**
+ * Helps provide tools to support recommended OpenMRS time zones conventions.
+ * 
+ * @see https://wiki.openmrs.org/display/docs/Time+Zones+Conventions
+ */
+public class TimeZoneUtil {
+
+	/**
+	 * Convert a date to the client timezone, and format it, to be readable to for the user. .
+	 *
+	 * @param date The date.
+	 * @param format the format to be used on the date
+	 * @return string with the date on the client timezone, formatted and ready to be displayed.
+	 */
+	public static String toClientTimezone(Date date, String format) {
+		String clientTimezone  = Context.getAuthenticatedUser().getUserProperty("clientTimezone");
+		return toClientTimezone(date, format,clientTimezone);
+
+	}
+
+	public static String toClientTimezone(Date date, String format, String clientTimezone) {
```

Comment:
This one should adopt appropriate naming and JavaDoc:
```java
/**
 * Formats a date while expressing it in the specified timezone.
 * ...
 * @param timezone The tz database name, eg. "Europe/Zurich".
 */
public static String toTimezone(Date date, String format, String timezone) {
```
About the **tz database**: https://en.wikipedia.org/wiki/Tz_database

Reply:
The java docs are on the [above](https://github.com/openmrs/openmrs-module-uiframework/pull/66/files/62e287b024f38792c889be92d1642c1726852e74#diff-7cfbeaf29655d32575647b73ea1fdef404134f18c7831b523722eb15439c6b48R34-R41), that one just overwrite that method, because sometimes we can use it without sending directly the clientTimezone